game-core/src/main/java/games/strategy/triplea/settings/ClientSetting.java

```diff
@@ -163,36 +170,23 @@
    * </p>
    */
   public static void initialize() {
-    setPreferences(Preferences.userNodeForPackage(ClientSetting.class));
-  }
-
-  @VisibleForTesting
-  static void setPreferences(final Preferences preferences) {
-    checkNotNull(preferences);
-
-    preferencesRef.set(preferences);
+    setPreferences(
+        Preconditions.checkNotNull(Preferences.userNodeForPackage(ClientSetting.class)));
   }
 
   @VisibleForTesting
   static void resetPreferences() {
     preferencesRef.set(null);
   }
 
-  private static Preferences getPreferences() {
-    final @Nullable Preferences preferences = preferencesRef.get();
-    if (preferences == null) {
-      throw new IllegalStateException("ClientSetting framework has not been initialized. "
-          + "Did you forget to call ClientSetting#initialize() in production code "
-          + "or ClientSetting#setPreferences() in test code?");
-    }
-    return preferences;
+  @Override
+  public void addSaveListener(final Consumer<String> saveListener) {
+    onSaveActions.add(saveListener);
```

Comment:
We might want to consider a `checkNotNull()` precondition here simply because this is a public API, and adding a `null` listener will trigger an NPE at some time in the future (during iteration) rather than when it was added.

Reply:
On the one hand it would mean we added new code that passed null, on the other hand we can get slightly better error detection. A null value would not work well for listener removal, and doing that instead of a no-op consumer is really quite wrong. Hopefully we'd see it during CR...

Reply:
At the cost of a bit of code clutter, the error detection is good to have: 817d7f121
Some point down the road we won't be passing null values anymore, we won't have to dedicate code to it anymore as any nulls would be all class-internal, any public facing nulls are either optional or calls the right API.